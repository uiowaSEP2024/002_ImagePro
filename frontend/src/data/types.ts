export type User = {
  first_name: string;
  last_name: string;
  email: string;
};

export type UserCreate = {
  email: string;
  password: string;
  first_name: string;
  last_name: string;
};

export type Provider = {
  id: number;
  name: string;
};

export type Job = {
  id: number;
  provider_job_name: string;
  customer_id: number;
  provider_job_id: string;
  provider_id: number;
  created_at?: string;
  num_steps?: number;
};

export type Key = {
  id: number;
  user_id: number;
  key: string;
  created_at?: string;
};

export type JobEvent = {
  kind: string;
  name: string;
  job_id: number;
  id: number;
  created_at?: string;
  event_metadata?: Record<string, any>;
};
