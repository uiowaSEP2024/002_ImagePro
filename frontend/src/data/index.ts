import { Job, JobEvent, Key, Provider, User, UserCreate } from "./types";


export const providers: Record<string, Provider> = {
  "2": {
    id: 2,
    name: "BotImage",
  },
};


export const jobs: Record<string, Job> = {
  "1": {
    id: 1,
    provider_job_name: "Kidney Cancer Detection",
    customer_id: 1,
    provider_job_id: "236",
    provider_id: 2,
    created_at: "2021-03-01T00:00:00.000Z",
  },

  "2": {
    id: 2,
    provider_job_name: "Lung Cancer Detection",
    customer_id: 1,
    provider_job_id: "237",
    provider_id: 2,
    created_at: "2021-03-01T00:00:00.000Z",
  },

  "3": {
    id: 3,
    provider_job_name: "Prostate Cancer Detection",
    customer_id: 1,
    provider_job_id: "238",
    provider_id: 2,
    created_at: "2021-03-01T00:00:00.000Z",
  },
};

export const events: JobEvent[] = [
  {
    id: 1,
    job_id: 5,
    kind: "Step",
    name: "Lung scan created",
    created_at: "2023-03-22T00:00:00.000Z",
  },
  {
    id: 2,
    job_id: 5,
    kind: "Step",
    name: "Lung scan started",
    created_at: "2023-03-23T00:00:00.000Z",
  },
  {
    id: 3,
    job_id: 5,
    kind: "Step",
    name: "Lung scan completed",
    created_at: "2023-03-24T00:00:00.000Z",
  },
];


export const fetchJobs = async (): Promise<Job[] | void> => {
  return fetch("http://localhost:8000/jobs", {
    credentials: "include",
    method: "GET",
  })
    .then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as Job[];
      }
    })
    .catch((e) => {
      console.log(e);
    });
};

// TODO: Replace with actual API call
export const fetchJobById = async (id: number): Promise<Job | void> => {
  return await fetch(`http://localhost:8000/jobs/${id}`, {
    credentials: "include",
    method: "GET",
  })
    .then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as Job;
      }
    })
    .catch((e) => {
      console.log(e);
    });
};

export const generateAPIKeys = async () => {
  await fetch("http://localhost:8000/api-keys", {
    credentials: "include",
    method: "POST"
  }).then((response) => response.json())
    .then((data) => {
      console.log(data)
      return data
    }).catch((e) => {
      console.log(e)
    })
}

export const fetchAPIkeys = async (): Promise<Key[] | void> => {
  return await fetch("http://localhost:8000/api-keys", {
    credentials: "include",
    method: "GET"
  }).
    then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as Key[];
      }
    })
    .catch((e) => {
      console.log(e)
    });
}


export const fetchEvents = async (
  jobId: number
): Promise<JobEvent[] | void> => {
  return await fetch(`http://localhost:8000/jobs/${jobId}/events`, {
    credentials: "include",
    method: "GET",
  })
    .then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as JobEvent[];
      }
    })
    .catch((e) => {
      console.log(e);
    });
};

export async function fetchCheckUserLoggedIn() {
  try {
    const result = await 
    fetch("http://localhost:8000/login", {
      credentials: "include",
      method: "GET",
    }) 
      
    return result.json() as unknown as {user?: User, message: string} 
  } catch (error) {
    console.log(error)
    throw error
  }
}


export const fetchLogout = async () => {
  const response = await fetch("http://localhost:8000/logout", {
    method: "POST",
    credentials: "include",
  })

  return response.json()
};


export const fetchLogin = async (email: string, password: string) => {
  const response = await fetch("http://localhost:8000/login", {
    credentials: "include",
    method: "POST",
    headers: {
      "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
    },
    body: new URLSearchParams({
      username: email,
      password: password,
    }),
  })

  // TODO: check for wider range of error codes
  if (response.status !== 200){
    throw new Error(await response.text())
  }

  return await response.json() as {user: User}
}




export const fetchSignUp = async (data: UserCreate) => {
  const response = await fetch("http://localhost:8000/users", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      email: data.email,
      password: data.password,
      first_name: data.first_name,
      last_name: data.last_name,
    }),
  })

  return await response.json()
}
