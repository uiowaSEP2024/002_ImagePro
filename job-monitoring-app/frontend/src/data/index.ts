import { Job, JobEvent, ApiKey, User, UserCreate } from "./types";

export const backendUrl = (
  process.env.NEXT_PUBLIC_BACKEND_URL || "http://localhost:8000"
).replace(/\/$/, "");

export const fetchJobs = async (): Promise<Job[] | void> => {
  return fetch(`${backendUrl}/jobs`, {
    credentials: "include",
    method: "GET"
  })
    .then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as Job[];
      }
    })
    .catch((e) => {
      console.log(e);
    });
};

// TODO: Replace with actual API call
export const fetchJobById = async (id: number): Promise<Job | void> => {
  return await fetch(`${backendUrl}/jobs/${id}`, {
    credentials: "include",
    method: "GET"
  })
    .then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as Job;
      }
    })
    .catch((e) => {
      console.log(e);
    });
};

export const generateAPIKeys = async () => {
  await fetch(`${backendUrl}/api-keys`, {
    credentials: "include",
    method: "POST"
  })
    .then((response) => response.json())
    .then((data) => {
      console.log(data);
      return data;
    })
    .catch((e) => {
      console.log(e);
    });
};

export const fetchAPIkeys = async (): Promise<ApiKey[] | void> => {
  return await fetch(`${backendUrl}/api-keys`, {
    credentials: "include",
    method: "GET"
  })
    .then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as ApiKey[];
      }
    })
    .catch((e) => {
      console.log(e);
    });
};

export const fetchGenAPIKeys = async (data: { note: string }) => {
  const response = await fetch(`${backendUrl}/api-keys`, {
    method: "POST",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  });

  return await response.json();
};

export const fetchEvents = async (
  jobId: number
): Promise<JobEvent[] | void> => {
  return await fetch(`${backendUrl}/jobs/${jobId}/events`, {
    credentials: "include",
    method: "GET"
  })
    .then(async (response) => {
      if (response.status == 200) {
        return (await response.json()) as JobEvent[];
      }
    })
    .catch((e) => {
      console.log(e);
    });
};

export async function fetchCheckUserLoggedIn() {
  try {
    const result = await fetch(`${backendUrl}/login`, {
      credentials: "include",
      method: "GET"
    });

    return result.json() as unknown as { user?: User; message: string };
  } catch (error) {
    console.log(error);
    throw error;
  }
}

export const fetchLogout = async () => {
  const response = await fetch(`${backendUrl}/logout`, {
    method: "POST",
    credentials: "include"
  });

  return response.json();
};

export const fetchLogin = async (email: string, password: string) => {
  const response = await fetch(`${backendUrl}/login`, {
    credentials: "include",
    method: "POST",
    headers: {
      "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8"
    },
    body: new URLSearchParams({
      username: email,
      password: password
    })
  });

  // TODO: check for wider range of error codes
  if (response.status !== 200) {
    throw new Error(await response.text());
  }

  return (await response.json()) as { user: User };
};

export const fetchSignUp = async (data: UserCreate) => {
  const response = await fetch(`${backendUrl}/users`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(data)
  });

  return await response.json();
};

export const fetchExpireApiKey = async (id: number) => {
  const response = await fetch(`${backendUrl}/api-keys/${id}/expire`, {
    credentials: "include",
    method: "POST"
  });

  return await response.json();
};

export const fetchDownloadReport = async (
  startDateStr: string,
  endDateStr: string
) => {
  console.log(startDateStr, endDateStr);
  const startDate = new Date(startDateStr);
  const endDate = new Date(endDateStr);

  startDate.setHours(new Date().getHours());
  startDate.setMinutes(new Date().getMinutes());
  startDate.setSeconds(new Date().getSeconds());
  startDate.setMilliseconds(new Date().getMilliseconds());

  endDate.setHours(new Date().getHours());
  endDate.setMinutes(new Date().getMinutes());
  endDate.setSeconds(new Date().getSeconds());
  endDate.setMilliseconds(new Date().getMilliseconds());

  const startTimestampSeconds = startDate.getTime() / 1000;
  const endTimestampSeconds = endDate.getTime() / 1000;

  console.log(startTimestampSeconds, endTimestampSeconds);

  const endpointUrl = `${backendUrl}/reporting?start_date=${startTimestampSeconds}&end_date=${endTimestampSeconds}`;

  console.log(endpointUrl);

  const response = await fetch(endpointUrl, {
    method: "GET",
    credentials: "include",
    headers: {
      "Content-Type": "text/csv"
    }
  });

  const blob = await response.blob();
  const url = window.URL.createObjectURL(blob);
  const a = document.createElement("a");
  a.href = url;
  a.download = `report-${startDateStr}-${endDateStr}.csv`;
  document.body.appendChild(a);
  a.click();
  a.remove();
  window.URL.revokeObjectURL(url);
};
