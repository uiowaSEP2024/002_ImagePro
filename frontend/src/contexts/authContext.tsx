import { fetchCheckUserLoggedIn, fetchLogin, fetchLogout } from "@/data";
import { User } from "@/data/types";
import { useRouter } from "next/router";
import React, { createContext, useEffect, useState } from "react";

interface IAuthContextProps {
  currentUser?: User;
  isAuthenticating: boolean;
  setCurrentUser: (user: any) => void;
  setIsAuthenticating: (loading: boolean) => void;
  logOut: () => Promise<void>;
  logIn: (email: string, password: string) =>  Promise<{user: User}|void>;
  refreshAuth: () => Promise<void>;
}

export const AuthContext = createContext<IAuthContextProps>({
  currentUser: undefined,
  isAuthenticating: true,
  setCurrentUser: () => {},
  setIsAuthenticating: () => {},
  logOut: async () =>{},
  logIn: async () =>{},
  refreshAuth: async () =>{}
});

export const AuthContextProvider = (props: React.PropsWithChildren) => {
  const [currentUser, setCurrentUser] = useState<User>();
  const [isAuthenticating, setIsAuthenticating] = useState(true);
  const router = useRouter();



  useEffect(()=>{
    refreshAuth()
  }, [])

  const logOut = async () => {
    try {
      await fetchLogout();
      await refreshAuth()
      router.push("/login");
    } catch (e) {
      throw e
    }
  };


  const logIn = async (email:string, password: string) => {
    let data
    try {
      data = await fetchLogin(email, password);
      await refreshAuth()
    } catch (e) {
      throw e
    }
    return data
  };

  const refreshAuth = async () =>{
    try{
      const data = await fetchCheckUserLoggedIn()
      setCurrentUser(data.user)
    }catch(e){
      setCurrentUser(undefined)
    }
    setIsAuthenticating(false)
  }


  return (
    <AuthContext.Provider
      value={{
        currentUser,
        isAuthenticating,
        setCurrentUser,
        setIsAuthenticating,
        logOut,
        logIn,
        refreshAuth
      }}
    >
      {props.children}
    </AuthContext.Provider>
  );
};
