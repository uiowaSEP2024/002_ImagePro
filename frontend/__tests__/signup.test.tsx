import {
  act,
  render,
  screen,
  waitFor,
  fireEvent
} from "@testing-library/react";
import Signup from "@/pages/signup";
import Billing from "@/pages/billing";
import Dashboard from "@/pages/dashboard";
import "@testing-library/jest-dom";
import { useRouter } from "next/router";
import { AuthContextProvider } from "@/contexts/authContext";

import * as data from "@/data";

const mockRouterPush = jest.fn();
jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      push: mockRouterPush
    };
  }
}));

jest.mock("@/data", () => ({
  fetchCheckUserLoggedIn() {
    return new Promise((resolve) => {
      resolve({ detail: "Not authenticated" });
    });
  },

  fetchSignUp() {
    return new Promise((resolve) => {
      const data = {
        email: "user@example.com",
        first_name: "string",
        last_name: "string",
        id: 0
      };
      resolve(data);
    });
  },

  fetchLogin() {
    return new Promise((resolve) => {
      const data = new URLSearchParams({
        email: "user@example.com",
        password: "abc"
      });
      resolve(data);
    });
  },

  fetchJobs() {
    return new Promise((resolve) => resolve([]));
  }
}));

// TODO: explore fixing snapshot testing with https://github.com/mui/material-ui/issues/21293#issuecomment-654921524
describe("SignUp", () => {
  it("renders text", async () => {
    await act(async () => render(<Signup />, { wrapper: AuthContextProvider }));
    expect(useRouter().push).not.toBeCalledWith("/");

    const text = await waitFor(() =>
      screen.getByRole("heading", {
        name: /Sign Up/i
      })
    );

    expect(text).toBeInTheDocument();
  });

  it("signup on change", async () => {
    await act(async () => render(<Signup />, { wrapper: AuthContextProvider }));

    await waitFor(() => {
      const firstNameInput = screen.queryByPlaceholderText(
        "First Name"
      ) as HTMLInputElement;
      fireEvent.input(firstNameInput, { target: { value: "John" } });
      const lastNameInput = screen.queryByPlaceholderText(
        "Last Name"
      ) as HTMLInputElement;
      fireEvent.input(lastNameInput, { target: { value: "Smith" } });
      const emailInput = screen.queryByPlaceholderText(
        "Email"
      ) as HTMLInputElement;
      fireEvent.input(emailInput, { target: { value: "abc@gmail.com" } });
      const passInput = screen.queryByPlaceholderText(
        "Password"
      ) as HTMLInputElement;
      fireEvent.input(passInput, { target: { value: "abc" } });
      const pass2Input = screen.queryByPlaceholderText(
        "Confirm Password"
      ) as HTMLInputElement;
      fireEvent.input(pass2Input, { target: { value: "abc" } });

      expect(firstNameInput.value).toBe("John");
      expect(lastNameInput.value).toBe("Smith");
      expect(emailInput.value).toBe("abc@gmail.com");
      expect(passInput.value).toBe("abc");
      expect(pass2Input.value).toBe("abc");
    });

    const button = await waitFor(() => screen.getByTestId("signup"));
    fireEvent.click(button);

    const notificationMessage = await waitFor(() =>
      screen.getByText(/Sign up successful/i)
    );

    await waitFor(() => {
      expect(notificationMessage).toBeInTheDocument();
    });
  });

  it("does not render internal pages", async () => {
    await act(async () =>
      render(<Billing />, { wrapper: AuthContextProvider })
    );

    expect(useRouter().push).toBeCalledWith("/login");

    await act(async () =>
      render(<Dashboard />, { wrapper: AuthContextProvider })
    );

    expect(useRouter().push).toBeCalledWith("/login");
  });
});
