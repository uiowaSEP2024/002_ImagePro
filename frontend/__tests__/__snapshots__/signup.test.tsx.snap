// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders Signup unchanged 1`] = `
<div>
  <div>
    <div
      class="nextui-c-fdHeMm nextui-c-fdHeMm-dwxLNB-responsive-true nextui-c-fdHeMm-iblJuSr-css"
    >
      <div
        class="nextui-c-BDLTQ nextui-c-jMTBsW nextui-c-gulvcB nextui-c-BDLTQ-kUADSW-variant-bordered nextui-c-BDLTQ-fmlsco-borderWeight-light nextui-c-BDLTQ-cuwTXc-disableAnimation-false nextui-c-BDLTQ-ilimyxf-css"
        role="section"
        tabindex="-1"
      >
        <p
          class="nextui-c-PJLV nextui-c-PJLV-hyvuql-weight-bold nextui-c-PJLV-ibsXxfh-css"
        >
          Sign Up
        </p>
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-ifGHEql-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-c-boZsAS-AmGUN-cv nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-gWufxv-underlined-true nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--underlined nextui-input-wrapper--shadow"
            >
              <span
                class="nextui-c-PJLV nextui-c-iFHsrU nextui-input-placeholder"
              >
                First Name
              </span>
              <input
                aria-label="First Name"
                aria-multiline="false"
                aria-placeholder="First Name"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria6005829987-1"
                placeholder="First Name"
                type="text"
                value=""
              />
              <button
                aria-label="clear"
                class="nextui-c-ctQnhE nextui-c-ctQnhE-kClpVX-underlined-true nextui-input-clear-button"
                lang="en"
                type="button"
              >
                <span
                  class="nextui-c-gDSmsE nextui-c-gulvcB"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-igvLppd-css"
        />
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-ifGHEql-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-c-boZsAS-AmGUN-cv nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-gWufxv-underlined-true nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--underlined nextui-input-wrapper--shadow"
            >
              <span
                class="nextui-c-PJLV nextui-c-iFHsrU nextui-input-placeholder"
              >
                Last Name
              </span>
              <input
                aria-label="Last Name"
                aria-multiline="false"
                aria-placeholder="Last Name"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria6005829987-3"
                placeholder="Last Name"
                type="text"
                value=""
              />
              <button
                aria-label="clear"
                class="nextui-c-ctQnhE nextui-c-ctQnhE-kClpVX-underlined-true nextui-input-clear-button"
                lang="en"
                type="button"
              >
                <span
                  class="nextui-c-gDSmsE nextui-c-gulvcB"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-igvLppd-css"
        />
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-idXLuPy-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-c-boZsAS-AmGUN-cv nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-gWufxv-underlined-true nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--underlined nextui-input-wrapper--shadow"
            >
              <span
                class="nextui-c-PJLV nextui-c-iFHsrU nextui-input-placeholder"
              >
                Email
              </span>
              <input
                aria-label="Email"
                aria-multiline="false"
                aria-placeholder="Email"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria6005829987-5"
                placeholder="Email"
                type="text"
                value=""
              />
              <button
                aria-label="clear"
                class="nextui-c-ctQnhE nextui-c-ctQnhE-kClpVX-underlined-true nextui-input-clear-button"
                lang="en"
                type="button"
              >
                <span
                  class="nextui-c-gDSmsE nextui-c-gulvcB"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-igvLppd-css"
        />
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-idXLuPy-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-c-boZsAS-AmGUN-cv nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-gWufxv-underlined-true nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--underlined nextui-input-wrapper--shadow"
            >
              <span
                class="nextui-c-PJLV nextui-c-iFHsrU nextui-input-placeholder"
              >
                Password
              </span>
              <input
                aria-label="Password"
                aria-multiline="false"
                aria-placeholder="Password"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria6005829987-7"
                placeholder="Password"
                type="text"
                value=""
              />
              <button
                aria-label="clear"
                class="nextui-c-ctQnhE nextui-c-ctQnhE-kClpVX-underlined-true nextui-input-clear-button"
                lang="en"
                type="button"
              >
                <span
                  class="nextui-c-gDSmsE nextui-c-gulvcB"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-igvLppd-css"
        />
        <div
          class="nextui-c-iwmZdB nextui-c-iwmZdB-jTlWbp-color-primary nextui-c-iwmZdB-jTfqsF-helperColor-default nextui-c-iwmZdB-dpkacN-borderWeight-normal nextui-c-iwmZdB-bkiCag-size-lg nextui-c-iwmZdB-idXLuPy-css nextui-input-main-container nextui-input-main-container--normal"
          data-state="normal"
        >
          <div
            class="nextui-c-boZsAS nextui-c-boZsAS-fjttfO-isTextarea-false nextui-c-boZsAS-iNKybL-animated-true nextui-c-boZsAS-AmGUN-cv nextui-input-container nextui-input-container--normal nextui-input-container--input"
          >
            <label
              class="nextui-c-eXOOPO nextui-c-eXOOPO-iNKybL-shadow-true nextui-c-eXOOPO-gWufxv-underlined-true nextui-c-eXOOPO-lkBNdM-isTextarea-false nextui-input-wrapper nextui-input-wrapper--normal nextui-input-wrapper--underlined nextui-input-wrapper--shadow"
            >
              <span
                class="nextui-c-PJLV nextui-c-iFHsrU nextui-input-placeholder"
              >
                Confirm Password
              </span>
              <input
                aria-label="Confirm Password"
                aria-multiline="false"
                aria-placeholder="Confirm Password"
                aria-readonly="false"
                aria-required="false"
                autocomplete="off"
                class="nextui-c-jeuecp nextui-input"
                data-state="normal"
                id="react-aria6005829987-9"
                placeholder="Confirm Password"
                type="text"
                value=""
              />
              <button
                aria-label="clear"
                class="nextui-c-ctQnhE nextui-c-ctQnhE-kClpVX-underlined-true nextui-input-clear-button"
                lang="en"
                type="button"
              >
                <span
                  class="nextui-c-gDSmsE nextui-c-gulvcB"
                >
                  <svg
                    class=""
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M12 2a10 10 0 1010 10A10.016 10.016 0 0012 2zm3.36 12.3a.754.754 0 010 1.06.748.748 0 01-1.06 0l-2.3-2.3-2.3 2.3a.748.748 0 01-1.06 0 .754.754 0 010-1.06l2.3-2.3-2.3-2.3A.75.75 0 019.7 8.64l2.3 2.3 2.3-2.3a.75.75 0 011.06 1.06l-2.3 2.3z"
                      fill="currentColor"
                    />
                  </svg>
                </span>
              </button>
            </label>
          </div>
          <div
            class="nextui-c-gJvMas nextui-input-helper-text-container"
          />
        </div>
        <div
          class="nextui-c-dfbgCO nextui-c-dfbgCO-fGHEql-fluid-true nextui-c-dfbgCO-igmUkpg-css"
        >
          <a
            class="nextui-c-kzikkm nextui-c-gulvcB nextui-c-kzikkm-iVlgeG-color-secondary nextui-c-kzikkm-fZgbCn-block-true nextui-c-kzikkm-isWIbj-animated-true nextui-c-kzikkm-jSBLRj-cv nextui-c-kzikkm-iclraoi-css nextui-link"
            href="/login"
          >
            Existing user? Log in. 
          </a>
        </div>
        <span
          aria-hidden="true"
          class="nextui-c-gNVTSf nextui-c-gNVTSf-hakyQ-inline-false nextui-c-gNVTSf-igvLppd-css"
        />
        <button
          class="nextui-c-iWjDFM nextui-c-gulvcB nextui-c-iWjDFM-hkKLfu-color-default nextui-c-iWjDFM-gkIlwi-size-md nextui-c-iWjDFM-cwXrJp-borderWeight-normal nextui-c-iWjDFM-iPJLV-css nextui-button nextui-button--ready"
          data-state="ready"
          type="button"
        >
          <span
            class="nextui-button-text"
          >
            Create Account
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
